src/browser/browser.zig

```diff
@@ -298,10 +302,23 @@ pub const Page = struct {
         const self: *Page = @ptrCast(@alignCast(ctx));
 
         log.debug("fetch module: specifier: {s}", .{specifier});
-        return try self.fetchData(
-            specifier,
-            if (self.current_script) |s| s.src else null,
-        );
+
+        const base = if (self.current_script) |s| s.src else null;
+
+        const file_src = blk: {
+            if (base) |_base| {
+                break :blk try URL.stitch(self.arena, specifier, _base);
+            } else break :blk specifier;
+        };
+
+        const slot = try self.module_map.getOrPut(self.arena, file_src);
+        if (slot.found_existing) {
+            return slot.value_ptr.*;
+        } else {
+            const module = try self.fetchData(specifier, base);
+            if (module) |_module| slot.value_ptr.* = _module;
```

Comment:
this is an problem, no?  You now have an entry in the map with an undefined value. If the same module is requested, `slot.found_existing` will be true, and you'll return undefined: `slot.value_ptr.*`